app/store/services/helpers/gps.js

```diff
@@ -0,0 +1,51 @@
+const getDistance = (
+  pointA = { longitude: 0, latitude: 0 },
+  pointB = { longitude: 0, latitude: 0 }
+) => {
+  const R = 6371e3; // metres
+  const φ1 = pointA.latitude * Math.PI / 180;
+  const φ2 = pointB.latitude * Math.PI / 180;
+  const Δφ = (pointB.latitude - pointA.latitude) * Math.PI / 180;
+  const Δλ = (pointB.longitude - pointA.longitude) * Math.PI / 180;
+
+  const a =
+    Math.sin(Δφ / 2) * Math.sin(Δφ / 2) +
+    Math.cos(φ1) * Math.cos(φ2) * Math.sin(Δλ / 2) * Math.sin(Δλ / 2);
+  const c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1 - a));
+
+  return R * c / 1000;
+};
+
+const isInArea = (
+  position = { longitude: 0, latitude: 0 },
+  point = { longitude: 0, latitude: 0 },
+  Δλ = 0,
+  Δφ = 0
+) =>
+  Math.abs(position.longitude - point.longitude) < Δλ &&
+  Math.abs(position.latitude - point.latitude) < Δφ;
+
+const getEventDistance = (points = []) => {
+  points.reduce((distance, currentPoint, index) => {
+    const nextPoint = points[index + 1];
+    if (nextPoint) {
+      return distance + getDistance(currentPoint, nextPoint);
+    }
+    return distance;
+  }, 0);
+};
+
+const getNearestPoint = (points = [], currentPosition = {}) => {
```

Comment:
findClosestNode -> you should is this formalism in naming 
https://www.youtube.com/watch?v=5RSMLgy06Ew&index=2&list=PL6MpDZWD2gTF3mz26HSufmsIO-COKKb5j

Reply:
gonna watch this!